apps/src/code-studio/components/progress/UnitCalendarLessonChunk.jsx

```diff
@@ -0,0 +1,142 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import {unitCalendarLessonChunk} from '@cdo/apps/templates/progress/unitCalendarLessonShapes';
+
+const styles = {
+  common: {
+    margin: 5,
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'center',
+    textAlign: 'center',
+    color: '#333',
+    textDecorationLine: 'none',
+    fontFamily: '"Gotham 4r", sans-serif'
+  },
+  assessment: {
+    border: '2px solid rgb(118, 101, 160)'
+  },
+  assessmentHover: {
+    border: '2px solid rgb(118, 101, 160)',
+    backgroundColor: 'rgb(118, 101, 160)',
+    color: 'white'
+  },
+  instructional: {
+    border: '2px solid #00adbc'
+  },
+  instructionalHover: {
+    border: '2px solid #00adbc',
+    backgroundColor: '#00adbc',
+    color: 'white'
+  },
+  isNotStart: {
+    borderLeftStyle: 'dashed'
+  },
+  isNotEnd: {
+    borderRightStyle: 'dashed'
+  },
+  titleText: {
+    width: '100%',
+    paddingRight: 15
+  },
+  iconSection: {
+    display: 'flex',
+    flexDirection: 'column',
+    justifyContent: 'space-between',
+    height: '100%',
+    paddingLeft: 2,
+    paddingTop: 2,
+    paddingBottom: 2,
+    boxSizing: 'border-box'
+  }
+};
+
+export default class UnitCalendarLessonChunk extends Component {
+  static propTypes = {
+    minuteWidth: PropTypes.number.isRequired,
+    lesson: unitCalendarLessonChunk,
+    isHover: PropTypes.bool,
+    handleHover: PropTypes.func.isRequired
```

Comment:
Its cool to see how your handled the hover here. Very thoughtful. In most of our React components we use an external to make this detection of hover easier. Its called Radium (https://formidable.com/open-source/radium/). I think using it here will simplify some of the extra work you had do to. Check out how it plays out in this example https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/templates/progress/LessonExtrasFlagIcon.jsx.

Reply:
This should allow you to simplify some of the logic you have written below for determining style and combine a couple of the styles above so I'll hold off on diving into those for now